### PR TITLE
Build component libraries

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
- Surfaced new command, `skyux build-public-library`.
- Bundles all files located in `src/app/public` as UMD module.
- After the bundle is successfully created, the pipeline is responsible for publishing to NPM.

**To test locally:**
- Clone the `feature-cleanup` branch from `skyux-template-library`: https://github.com/blackbaud/skyux-template-library/tree/feature-cleanup
- Type `npm install && skyux build-public-library`

### CODE REVIEW

**Comment by Blackbaud-SteveBrush** [comment]
@Blackbaud-BobbyEarl I still need to finish up the unit tests for this branch, but do you mind running through it quickly to make sure there aren't any glaring issues with my approach?

**Comment by Blackbaud-BobbyEarl** [comment]
Really the only thing that stands out, is the Travis shell scripts.  Those make sense for our open source libraries, but not sure they'll be the same technique we use to publish to the VSTS registry through our pipeline.  We may need to regroup on that aspect of things.

**Comment by Blackbaud-SteveBrush** [comment]
@Blackbaud-BobbyEarl This is ready for an official review.

File: cli/build-public-library.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  };
+
+  fs.writeJSONSync(skyPagesConfigUtil.spaPathTemp('tsconfig.json'), config);
+}
+
+function transpile(skyPagesConfig) {
+  const config = webpackConfig.getWebpackConfig(skyPagesConfig);
+  const compiler = webpack(config);
+
+  return new Promise((resolve, reject) => {
```
Comment (Blackbaud-BobbyEarl) [inline cli/build-public-library.js]: I wonder if https://github.com/blackbaud/skyux-builder/blob/master/cli/build.js#L133-L163 is similar enough to factor this functionality out?

File: index.js [hunk 2]
```diff
@@ -13,6 +13,9 @@ module.exports = {
       case 'build':
         require('./cli/build')(argv, skyPagesConfig, webpack);
         break;
+      case 'build-public-library':
+        require('./cli/build-public-library')(skyPagesConfig);
```
Comment (Blackbaud-BobbyEarl) [inline index.js]: I probably originally did it for testing, but should we pass in webpack here to stay consistent?

**Pull request merged**